src/Microsoft.Health.Dicom.CosmosDb/CosmosDbQueryGenerator.cs

```diff
@@ -0,0 +1,140 @@
+
+            // get the tag
+            var tagName = FormatTagString(rangeValueMatchCondition.QueryTag);
+
+            var fromDate = rangeValueMatchCondition.Minimum;
+            var toDate = rangeValueMatchCondition.Maximum;
+
+            // match on formatting written to cosmos
+            var fromDateFormatted = fromDate.ToString("yyyyMMdd");
+            var toDateFormatted = toDate.ToString("yyyyMMdd");
+
+            string condition = $"(c['value']['{tagName}']['Value'][0] BETWEEN '{fromDateFormatted}' AND '{toDateFormatted}')";
+
+            AddToQuery(condition);
+        }
+
+        public override void Visit(DateSingleValueMatchCondition dateSingleValueMatchCondition)
+        {
+            // Ensure the condition is not null
+            EnsureArg.IsNotNull(dateSingleValueMatchCondition, nameof(dateSingleValueMatchCondition));
+
+            // get the tag and convert the value to match the date format on the dicom storage
+            var tagName = FormatTagString(dateSingleValueMatchCondition.QueryTag);
+            string tagValue = dateSingleValueMatchCondition.Value.ToString("yyyyMMdd");
+            string condition = $"(c['value']['{tagName}']['Value'][0] = '{tagValue}')";
+
+            AddToQuery(condition);
+        }
+
+        public override void Visit(PersonNameFuzzyMatchCondition fuzzyMatchCondition)
+        {
+            // Ensure the condition is not null
+            EnsureArg.IsNotNull(fuzzyMatchCondition, nameof(fuzzyMatchCondition));
+            var tagName = FormatTagString(fuzzyMatchCondition.QueryTag);
+            var tagValue = fuzzyMatchCondition.Value.ToString();
+
+            // Break up the string into tokens of names, and do regex matching on each part
+            var nameWords = tagValue.Split(' ');
+            var nameWordsLength = nameWords.Length;
+            for (var i = 0; i < nameWordsLength; i++)
+            {
+                nameWords[i] = $"{nameWords[i]}.*";
+            }
+            var condition = $"(REGEXMATCH(c['value']['{tagName}']['Value'][0]['Alphabetic'], '{String.Join("", nameWords)}', 'i'))";
+
+            AddToQuery(condition);
+        }
+
+        public override void Visit(DoubleSingleValueMatchCondition doubleSingleValueMatchCondition)
+        {
+            // Ensure the condition is not null
+            EnsureArg.IsNotNull(doubleSingleValueMatchCondition, nameof(doubleSingleValueMatchCondition));
+            // get the tag & value
+            var tagName = FormatTagString(doubleSingleValueMatchCondition.QueryTag);
+            var tagValue = doubleSingleValueMatchCondition.Value;
+            var condition = $"(c['value']['{tagName}']['Value'][0] = {tagValue})";
+
+            AddToQuery(condition);
+        }
+
+        public override void Visit(LongRangeValueMatchCondition longRangeValueMatchCondition)
+        {
+            throw new NotImplementedException();
```

Comment:
is there a plan to eventually implement this? or will it always be unimplemented? Wondering if there should be a `TODO`

Reply:
This one doesn't make sense to implement given the syntax used in the other ranges. It currently doesn't know what to do. IE, if you wanted everything before a value of -10, the format would mimic `someAttribute=--10`. But, definitely should include a note.

Reply:
Oh I see! thanks for explaining, that makes a lot of sense